portfolio/src/main/webapp/gallery.js

```diff
@@ -0,0 +1,48 @@
+
+var travelSlideshow = {
+  slideIndex: 0,
+  plusSlides: function(offset) {
+    this.showSlides(this.slideIndex += offset);
+  }, 
+  currentSlide: function(newSlideIndex) {
+    this.showSlides(this.slideIndex = newSlideIndex);
+  },
+  hideAllSlides: function(slides) {
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";  
+    }
+  },
+  disableAllDots: function(dots) {
+    for (i = 0; i < dots.length; i++) {
+        dots[i].className = dots[i].className.replace(" active", "");
+    }
+  },
+  showSlides: function(activatedSlide) {
+    var i;
+    var slides = document.getElementsByClassName("slides");
+    var dots = document.getElementsByClassName("dot");
+
+    // Keeps counter in the proper interval
+    if (activatedSlide >= slides.length) {
+      this.slideIndex = 0;
+    }  
+    if (activatedSlide < 0) {
+      this.slideIndex = slides.length - 1;
+    }
+
+    this.hideAllSlides(slides);
+    this.disableAllDots(dots);
+
+    // Make current photo visible
+    console.log(this.slideIndex);
+    slides[this.slideIndex].style.display = "block";  
```

Comment:
What happens if there are no slides at all? The code should still do something reasonable.